From c7318444b811125f26828fd39e8a46de81cd5f86 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 7 Aug 2013 09:13:11 -0400
Subject: [PATCH 7/8] Apparently if something goes wrong on the HSM, we wind up
 with 0-size.

Handle zero-sized output by erroring in the rpm macro.  Eventually we
should make sure pesign is throwing an error there too.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/macros.pesign | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/macros.pesign b/src/macros.pesign
index 84e87a3..6b22826 100644
--- a/src/macros.pesign
+++ b/src/macros.pesign
@@ -47,5 +47,8 @@
     elif [ -n "%{-i*}" -a -n "%{-e*}" ]; then				\
       touch %{-e*}							\
     fi									\
+  fi									\
+  if [ ! -s %{-o} ]; then						\
+    exit 1								\
   fi ;
 
-- 
1.8.3.1

